packages/extension-ui/src/Popup/CreateAccount/index.tsx

```diff
@@ -23,10 +23,18 @@ function CreateAccount ({ className }: Props): React.ReactElement {
   const [isBusy, setIsBusy] = useState(false);
   const [step, setStep] = useState(1);
   const [account, setAccount] = useState<null | { address: string; seed: string }>(null);
-  const type = DEFAULT_TYPE;
+  const [type, setType] = useState(DEFAULT_TYPE);
   const [name, setName] = useState('');
   const options = useGenesisHashOptions();
-  const [genesis, setGenesis] = useState('');
+  const [genesisHash, setGenesis] = useState('');
+
+  function getChainName (genesisHash: string): string {
+    const filtered = options.filter(({ value }) => {
+      return genesisHash === value;
+    });
+
+    return filtered.length > 0 ? filtered[0].text : '';
+  }
```

Comment:
Just move this to file level, i.e. outside the React component. As tis stands the function would be re-created on each render.

Reply:
I actually removed the function for simplicity. I think it will require a better design to support it if more chains are going to support Ethereum